backend/functions/src/resources/user-profile/user-profile.controller.ts

```diff
@@ -0,0 +1,18 @@
+import { RequestHandler } from 'express';
+import { UserProfile } from '.';
+import { getUserProfileService, patchUserProfileService } from '../../services/user-profile.service';
+
+export const getUserProfile: RequestHandler = async (req, res) => {
+  if (typeof req.params.id === 'string') {
```

Comment:
Route and query parameters are always strings. There is no point of having this check. Remove this.